server/form-pages/utils/defaultMatchingInformationValues.test.ts

```diff
@@ -0,0 +1,130 @@
+import { retrieveOptionalQuestionResponseFromFormArtifact } from '../../utils/retrieveQuestionResponseFromFormArtifact'
+import { assessmentFactory } from '../../testutils/factories'
+import MatchingInformation, {
+  MatchingInformationBody,
+} from '../assess/matchingInformation/matchingInformationTask/matchingInformation'
+import { defaultMatchingInformationValues } from './defaultMatchingInformationValues'
+
+jest.mock('../../utils/retrieveQuestionResponseFromFormArtifact')
+
+describe('defaultMatchingInformationValues', () => {
+  const bodyWithUndefinedValues: MatchingInformationBody = {
+    acceptsChildSexOffenders: undefined,
+    acceptsHateCrimeOffenders: undefined,
+    acceptsNonSexualChildOffenders: undefined,
+    acceptsSexOffenders: undefined,
+    apType: undefined,
+    cruInformation: undefined,
+    hasEnSuite: undefined,
+    isArsonDesignated: undefined,
+    isArsonSuitable: undefined,
+    isCatered: undefined,
+    isSingle: undefined,
+    isStepFreeDesignated: undefined,
+    isSuitableForVulnerable: undefined,
+    isSuitedForSexOffenders: undefined,
+    isWheelchairDesignated: undefined,
+    lengthOfStay: undefined,
+    lengthOfStayAgreed: undefined,
+    lengthOfStayDays: undefined,
+    lengthOfStayWeeks: undefined,
+    specialistSupportCriteria: undefined,
+  }
+  const assessment = assessmentFactory.build()
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('returns an object with current or sensible default values for relevant fields', () => {
+    ;(retrieveOptionalQuestionResponseFromFormArtifact as jest.Mock).mockReturnValue('yes')
+
+    const body: MatchingInformationBody = {
+      ...bodyWithUndefinedValues,
+      lengthOfStayAgreed: 'no',
+      lengthOfStayDays: '3',
+      lengthOfStayWeeks: '3',
+    }
+
+    expect(defaultMatchingInformationValues(MatchingInformation, body, assessment)).toEqual({
+      isWheelchairDesignated: 'essential',
+      lengthOfStay: '24',
+    })
+  })
+
+  describe('isWheelchairDesignated', () => {
+    it('is set to the original value if defined', () => {
+      expect(
+        defaultMatchingInformationValues(
+          MatchingInformation,
+          { ...bodyWithUndefinedValues, isWheelchairDesignated: 'desirable' },
+          assessment,
+        ),
+      ).toEqual(expect.objectContaining({ isWheelchairDesignated: 'desirable' }))
+    })
+
+    it("is set to 'essential' when there's no original value and `needsWheelchair` === 'yes'", () => {
+      ;(retrieveOptionalQuestionResponseFromFormArtifact as jest.Mock).mockReturnValue('yes')
+
+      expect(defaultMatchingInformationValues(MatchingInformation, bodyWithUndefinedValues, assessment)).toEqual(
+        expect.objectContaining({ isWheelchairDesignated: 'essential' }),
+      )
+    })
+
+    it("is set to 'notRelevant' when there's no original value and `needsWheelchair` === 'no'", () => {
+      ;(retrieveOptionalQuestionResponseFromFormArtifact as jest.Mock).mockReturnValue('no')
```

Comment:
again, do you thiknk it's important to be clear which question's answer we are stubbing as `'no'` here? To understand our mapping clearly I feel like I need to know from the mock that it's the answer to the question `needsWheelchair`...